### ISSUE TITLE
Multiple persistent shareable scenarios

### ISSUE THREAD

**Issue by gj262** [neherlab/covid19_scenarios#181]
# 🙋 Feature Request

The current implementation shows a single scenario and allows the user to change it and view the results. The scenario input can be shared through the app URL.

This feature allows for multiple persistent and shareable scenarios as the title suggests :).

## 🔦 Context

This feature evolves from issues #11 and #31. And while not implementing these precisely it makes progress in that direction and evolves to meet the items in some of comments in those issues. Hence a new feature!

## 😯 Describe the feature

There will be tabs below the main header, one to customize scenarios and one for each saved scenario. 

## 💁 Possible Solution

Initially users will see no tabs <sup>1</sup>. There will be a **save action** that is clickable once inputs are modified. Clicking save will present a modal requesting a name and owner email for the scenario to be saved. Once saved, the scenario (name, owner, input, result) will persist to local storage. And the view will change to tabs <sup>1</sup>. The in view tab is titled "Customize" with the prior scenario state and a new tab will appear for the saved scenario titled with that scenarios name . 

The view of the saved scenario will include a **delete action** protected with an are you sure modal.

There will be a **share action or link** in each tab which provides the saved scenario (name, owner, input, result) as a URL.

If a user clicks on a shared URL the application state will show their own saved scenario tabs plus an extra tab with this shared scenario titles name + (owner). They can choose to save this shared scenario or close the tab or delete it.

As a first pass, saved scenarios should be immutable i.e. the inputs are not changeable and are disabled in the saved scenario view. Otherwise there is a possibility for confusion when scenarios are shared and then edited by various participants. Saved scenarios could be cloned and the clones input could be changed. <sup>2</sup>.

Initially the actions would follow the button format currently in use for Run and Export. However this should be improved with a separate issue.

URL state will match the in view tab scenario.

## Future 

There is now a local store of named and owned scenarios. This could be stored remotely. There could be a page to manage these scenarios. There could be views to compare the results.

## To Be Resolved

- [ ] Tabs will not work for mobile
- [ ] Can a result be URL serialized?

## Notes

1. Having tabs appear for the first saved scenario is lame UX. But the alternative of always having a single "Customize" tab is also somewhat lame. Will need to experiment.
2. Follow some existing art for best practices.

**Comment by rcbevans** [comment]
One possibility would be to have an accordion stack of customize scenario cards within a new top level "scenario's" card.  The top level scenarios card would have an add button, which would collapse the currently expanded scenario, and add a new one ready to customize.  The "Scenario" header could be modifiable to name each scenario being worked on.  A previous scenario could then be switch to by tapping on the scenario card header to expand it, collapsing the current one.  This UI would also be more mobile friendly.

**Comment by SharanBabu** [comment]
I have been working on a design. I understand you already have implemented this using tabs. But in this design, I have created a separate page for all 'Saved scenarios' and have rendered the scenarios as '[Adaptive cards](https://docs.microsoft.com/en-us/adaptive-cards/sdk/rendering-cards/javascript/render-a-card)' which I think will be better for mobile. Let me know what you think.

![image](https://user-images.githubusercontent.com/8034623/77858484-d7671680-71b8-11ea-8912-a013846769b3.png)

**Comment by gj262** [comment]
Hi @SharanBabu, 

I like that. I see the work I'm doing in this PR as a foundation for whatever UX best fits the core Neherlab workflows. https://github.com/neherlab/covid19_scenarios/issues/11#issuecomment-600076600 for instance talks about being able to compare scenarios which would be something to incorporate into your Save Scenarios view. 

One feature I like about Tabs is that I can quickly switch back and forth between slightly tweaked scenarios. **But** I'm not the target user, so that may not be a feature that the scientists care about / would find useful.

**Comment by SharanBabu** [comment]
Ok, once your PR is merged, I will start working on incorporating this view. I agree with you on the ability to switch back and forth between scenarios easily with tabs. But I think the user can still do the switching by opening multiple scenarios in multiple browser tabs (or even side by side with multiple windows).

**Comment by SharanBabu** [comment]
@gj262 @ivan-aksamentov  I have submitted a [PR](https://github.com/neherlab/covid19_scenarios/pull/421) for this feature. The work is not code complete, but would love some early feedback on it.

**Comment by gj262** [comment]
I'm going to close this as it is not something I've heard an immediate desire for from the NeherLab/Community.
